ts/IterableInbox.tsx

```diff
@@ -7,18 +7,40 @@ import IterableInboxEmptyState from './IterableInboxEmptyState'
 import sampleMessages from './sampleMessageData.js'
 
 const IterableInbox = () => {
-   const message = "Inbox";
+   const inboxTitle = "Inbox";
+   const [selectedMessageIdx, setSelectedMessageIdx] = useState(null);
    const [messages, setMessages] = useState(sampleMessages);
 
+   function handleMessageSelect(index: number) {
+      let updatedMessages = messages;
+      let updatedMessage = messages[index];
+
+      if(!updatedMessage.read) {
+         updatedMessage.read = true
+      }
+
+      updatedMessages[index] = updatedMessage;
+      setMessages([...updatedMessages]);
+      setSelectedMessageIdx(index);
+   }   
```

Comment:
small nit: 
personally I find it easier to follow what exactly the impact is of a function, when all the required components are passed in as part of the function's properties. given that you are relying on `messages` as an outside requirement, I find it easier to understand what's going on when the function contract looks like:

```
const handleMessageSelect = (index, messages) => {  
  ....code code code...
}
```
This has the added benefit that you don't really need to include the `let` statements since you're passing the messages in.


down on line `36` you can update the line to read: `handleMessageSelect={(index) => handleMessageSelect(index, messages)}`

what do you think?

Reply:
Good point, Patrick. I think it is better design to have the messages sent into the function as a parameter. I will make the change and send a new commit.